tasks/lzards-backup/src/index.ts

```diff
@@ -0,0 +1,277 @@
+      log.error(`${granule.granuleId}: Granule did not have a properly defined collection and version, or refer to a collection that does not exist in the database`);
+      log.error(`${granule.granuleId}: Granule (${granule.granuleId}) will not be backed up.`);
+      error.message = `${granule.granuleId}: ${error.message}`;
+      throw error;
+    }
+  }
+
+  try {
+    const backupFiles = granule.files.filter(
+      (file) => shouldBackupFile(file.name, granuleCollection)
+    );
+    log.info(`${JSON.stringify(granule)}: Backing up ${JSON.stringify(backupFiles)}`);
+    return Promise.all(backupFiles.map((file) => makeBackupFileRequest({
+      creds,
+      authToken,
+      file,
+      collection: `${granule.dataType}___${granule.version}`,
+      granuleId: granule.granuleId,
+    })));
+  } catch (error) {
+    error.message = `${granule.granuleId}: ${error.message}`;
+    throw error;
+  }
+};
+
+export const generateAccessCredentials = async () => {
+  const params = {
+    RoleArn: getRequiredEnvVar('backup_role_arn'),
+    DurationSeconds: CREDS_EXPIRY_SECONDS,
+    RoleSessionName: `${Date.now()}`,
+  };
+  const roleCreds = await sts().assumeRole(params).promise();
+  return roleCreds as AWS.STS.AssumeRoleResponse;
+};
+
+export const getAuthToken = async () => {
+  const api = getRequiredEnvVar('launchpad_api');
+  const passphrase = await getSecretString(getRequiredEnvVar('launchpad_passphrase_secret_name'));
+  if (!passphrase) {
+    throw new GetAuthTokenError('The value stored in "launchpad_passphrase_secret_name" must be defined');
+  }
+  const certificate = getRequiredEnvVar('launchpad_certificate');
+  const token = await getLaunchpadToken({
+    api, passphrase, certificate,
+  });
+  return token;
+};
+
+export const backupGranulesToLzards = async (event: HandlerEvent) => {
+  // Given an array of granules, submit each file for backup.
+  log.warn(`Running backup on ${JSON.stringify(event)}`);
+  const roleCreds = await generateAccessCredentials();
+  const authToken = await getAuthToken() as string;
+
+  const backupPromises = (event.input.granules.map(
+    (granule) => backupGranule(roleCreds, authToken, granule)
+  ));
+
+  const backupResults = await Promise.allSettled(backupPromises);
+
+  // If there are uncaught exceptions, we want to fail the task.
+  if (backupResults.some((result) => result.status === 'rejected')) {
+    log.error('Some LZARDS backup results failed due to internal failure');
+    log.error('Manual reconciliation required - some backup requests may have processed');
+    log.error(`Full output: ${JSON.stringify(backupResults)}`);
+    throw new Error(`${JSON.stringify(backupResults)}`);
+  }
+  const filteredResults = backupResults.filter(
+    (result) => isFulfilledPromise(result)
+  ) as PromiseFulfilledResult<makeBackupFileRequestResult[]>[];
+  return {
+    backupResults: filteredResults.map((result) => result.value).flat(),
+    originalPayload: event.input,
```

Comment:
Hmm, if we do that, where would you suggest we stash `backupResults`?

Reply:
I would leave backupResults as a top level key on the returned output. The output would be:

```
{
    backupResults: filteredResults.map((result) => result.value).flat(),
    granules: event.input.granules,
}
```

Reply:
Oh, I see, I misunderstood. 

I suppose the *entire point* of structuring inputs/outputs in the workflow configuration is that we don't have to write logic like that.     I was approaching it from the `whatever payload you give this task, it will take and return the granules object, but that's not in-line with our other tasks. 

👍🏻  

I updated the schemas/outputs to match: 

6438eb31f